serpentTools/parsers/microxs.py

```diff
@@ -0,0 +1,271 @@
+"""Parser responsible for reading the ``*mdx[i].m`` files"""
+import re
+
+from numpy import array
+
+from serpentTools.engines import KeywordParser
+from serpentTools.objects import splitItems
+from serpentTools.objects.readers import BaseReader
+
+from serpentTools.messages import SerpentToolsException
+
+
+class MicroXSReader(BaseReader):
+    """
+    Parser responsible for reading and working with micro-xs (mdx) files.
+
+    Attributes
+    ------------
+    nfy: dict
+        Nested dictionary with tuples (parent, energy) as keys.
+        Parent is the isotope undergoing fission, e.g. 922350
+        and energy is the impending neutron energy causing fission
+        in MeV.
+        The values are nested dictionaries with the following structure:
+            'fissProd': array
+                contains the fission product ids
+                e.g., [541350, 551350, ... ]
+            'indYield': array
+                contains the independent yields
+                e.g., [0.053, 0.015, ... ]
+            'cumYield': array
+                contains the cumulative yields
+                e.g., [0.060, 0.016, ... ]
+    fluxRatio: dict
+        Dictionary with universes id as keys and the
+        corresponding group-wise flux values.
+        e.g., fluxRatio['0']= [9.91938E+14, 1.81729E+15, ...]
+    fluxUnc: dict
+        Dictionary with universes id as keys and the
+        corresponding group-wise flux uncertainty values.
+        e.g., fluxRatio['0']= [0.00023, 0.00042, ...]
+    xsVal: dict
+        Nested dictionary with universes as as keys, e.g. '0'.
+        The values are nested dictionary with tuples as keys
+        (isotope, reaction, flag) and group xs as values.
+        e.g., (922350, 18, 0)
+    xsUnc: dict
+        Nested dictionary with universes as as keys, e.g. '0'.
+        The values are nested dictionary with tuples as keys
+        (isotope, reaction, flag) and group xs uncertainties.
+
+    Parameters
+    ----------
+    filePath: str
+        path to the *mdx[n].m file
+
+    Raises
+    ------
+    SerpentToolsException:
+                            No results exist in the file
+                            No results are collected
+    IOError: file is unexpectedly closes while reading
+    """
+
+    def __init__(self, filePath):
+        BaseReader.__init__(self, filePath, 'microxs')
+        self._energyFY = []
```

Comment:
I mention this below, but is this something that should be stored publicly, or just a variable local to the `_storeFissionYields` method?

Reply:
Should be stored.